library/core/src/fmt/mod.rs

```diff
@@ -311,6 +311,7 @@ impl<'a> Arguments<'a> {
     #[inline]
     #[unstable(feature = "fmt_internals", reason = "internal to format_args!", issue = "none")]
     pub fn new_v1(pieces: &'a [&'static str], args: &'a [ArgumentV1<'a>]) -> Arguments<'a> {
+        assert!((0..=args.len() + 1).contains(&pieces.len()));
```

Comment:
Aren't the actual constraints here just that pieces is either the same size or one element larger than args?

Reply:
Yep, not sure why I put `0`...